docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,124 @@
+
+Vertical Lagrangian-remap method
+======================
+
+date: 2020/11/24
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+
+
+Summary
+-------
+
+The vertical Lagrangian-remap method is to be implemented to support the choice 
+of hybrid vertical coordinates. It can be seen as an extension of the Arbitrary 
+Lagrangian-Eulerian method which is already implemented in MPAS-Ocean, a key 
+difference being that the solution of diasurface transport is coordinate-free. 
+This has the advantage of removing the vertical CFL constraint, which is of 
+particular concern for an evolving ice shelf geometry in terrain-following 
+coordinate schemes. Generalization of the vertical coordinate in the horizontal 
+equations of motion is not in the scope of this development. This development 
+will include some infrastructure for specifying the target grid for regridding. 
+The minimum criteria for success are not degrading the accuracy of the solution 
+or the performance by more than [an acceptable threshold] for several basic test 
+cases... .
+
+*The purpose of this section is to summarize what capability is to be added to
+MPAS-Model through this design process. It should be clear what new code will do
+that the current code does not. Summarizing the primary challenges with respect
+to software design and implementation is also appropriate for this section.
+Finally, this statement should contain a general statement with regard to what
+is "success."*
+
+
+Requirements
+------------
+
+Requirement: name-of-requirement-here
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/11/24
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Requirement: A grid that relaxes to an analytical expression for a target grid.
+
+Requirement: The ability to implement hybrid coordinate systems.
+
+Requirement: XX order accuracy of the solution.
+
+Requirement: Performance is not degraded below an acceptable level.
+
+Requirement: Layer thicknesses are non-negative (and above a minimum thickness).
+
+*Possible additional requirements:
+- re. development of instabilities
```

Comment:
I think it goes without saying that we don't want to cause numerical instabilities but if there are specific instabilities that this approach is at risk of causing, it would certainly be worth requiring that they be avoided.

We do want the approach to be "robust" in the sense that we'd like it to support quite a variety of vertical coordinate choices without causing numerical instabilities.  But some of that may be beyond the scope of just vertical remapping (e.g. relating to horizontal pressure-gradient force).